portal-kernel/src/com/liferay/portal/kernel/util/SubscriptionSender.java

```diff
@@ -154,6 +161,30 @@ public void flushNotifications() throws Exception {
 
 			_persistestedSubscribersOVPs.clear();
 
+			for (ObjectValuePair<String, Long> ovp :
```

Comment:
This logic seems to be identical to the non-staging case (other than the boolean). I think it is unnecessary to make the distinction.

Instead of adding a new addPersistedStagingSubscribers() maybe we should rename it to addPersistedDelayedSubscribers() (or something like that, maybe adding an additional boolean value to the regular addPersistedSubscribers() is better) to avoid tying this to staging.